### PR TITLE
[ARM] Az Deployments : Add support for --query-string

### PR DESCRIPTION
**Pull request by detienne20**
PR is dependent on https://github.com/Azure/azure-cli/pull/16179. 

**Description**

Az deployments now supports  --query-string to be used with --template-uri for linked template deployments; --query-string is a SAS token for a storage container that contains the main template and linked template(s).  Support for the --query-string parameter will facilitate the transition to template specs. Like with template specs linked templates are referenced through the relative path field in template link as illustrated below:
      
        "templateLink":  {
                  "relativePath":"linked_template.json",
                   "contentVersion":"1.0.0.0"
         }

**Testing Guide**

1. Create storage account container and upload files for deployment. 

     az storage account create -n {} -g {} -l {} --sku {} --kind {} 
     az storage container create -n {} --account-name {}
     az storage blob upload -c {} -f "{}" -n {} --account-name {}

2. Generate an SAS token for the entire container (query-string parameter) 

     az storage container generate-sas --account-name {} --account-key {} --name {} --permissions dlrw --expiry {}

3. Generate the url for the main template (template-uri parameter) 

     az storage blob url -c {} -n {} --account-name {}

4. Deploy with --template-uri and --query-string parameter. 

     az deployment group create -g {} --template-uri {blob_url} --query-string "{sas_token}"
     az deployment sub create -g {} --template-uri {blob_url} --query-string "{sas_token}"
     az deployment mg validate  -g {} --template-uri {blob_url} --query-string "{sas_token}"
     az deployment tenant validate  -g {} --template-uri {blob_url} --query-string "{sas_token}"

### CODE REVIEW

**Comment by yonzhan** [comment]
ARM

File: src/azure-cli/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -852,8 +862,16 @@ def _prepare_deployment_properties_unmodified(cmd, template_file=None, template_
     template_obj = None
     on_error_deployment = None
     template_content = None
+
+    if query_string and not template_uri:
+        raise IncorrectUsageError('must provide --template-uri if --query-string is specified')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/custom.py:867]: Please make the tone more friendly
```suggestion
        raise IncorrectUsageError('please provide --template-uri if --query-string is specified')
```

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 2]
```diff
@@ -1012,6 +1012,99 @@ def test_export_template_spec(self, resource_group, resource_group_location):
 
 
 
+class DeploymentTestsWithQueryString(LiveScenarioTest):
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1015]: May I ask why are they live only tests?

Comment (detienne20) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1015]: The tests generate SAS tokens which fail the cred scan, so I have changed them to live tests.

**Comment by detienne20** [comment]
Created new PR https://github.com/Azure/azure-cli/pull/16447 without extra reviewers/changes.

**Pull request closed without merging**